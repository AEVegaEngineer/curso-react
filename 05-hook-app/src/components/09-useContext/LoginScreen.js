import React, {useContext} from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    //1 obtener la referencia al context\    
    //2 setuser
    //3 
    const user = {
        id: 123,
        name: 'Andres'
    }
    const {setUser} = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr/>
            <button className="btn btn-primary" onClick={() => {setUser(user)}}>
                Login
            </button>
        </div>
    )
}
