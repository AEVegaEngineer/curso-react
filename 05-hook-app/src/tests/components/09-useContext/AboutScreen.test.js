import React from 'react'
import {mount, configure} from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { UserContext } from '../../../components/09-useContext/UserContext';
import { AboutScreen } from '../../../components/09-useContext/AboutScreen';

configure({adapter: new Adapter()});


describe('Pruebas en <AboutScreen/>', () => {
  // para poder usar el AboutScreen se debe crear un contexto, basicamente
  // esto funciona creando un contexto en UserContext.js
  // luego MainApp.js establece ese context con un provider y coloca en su value
  // las variables que se quieren publicar en ese contexto y debajo del contexto
  // pone el Router y como la app esta dentro de ese router, tiene acceso a esas variables
  const user = {
    name: 'Andres',
    email: 'aevega1991@gmail.com'
  }
  const setUser = jest.fn();
  // si usamos shallow, se va a renderizar unicamente el higher order component
  // es decir que toda la informacion subyacente del componente queda oculta
  // si necesitamos que se vea lo que contiene el componente
  const wrapper = mount(
    <UserContext.Provider value={{ user, setUser }} >
      <AboutScreen/>
    </UserContext.Provider>
  );
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar al usuario en el tag pre', () => {
    expect(JSON.parse(wrapper.find('pre').text().trim())).toEqual(user);    
    expect( setUser ).toHaveBeenCalledTimes(0);
  });

  test('debe de desloguear al usuario al hacer click en logout', async() => {
    expect( setUser ).toHaveBeenCalledTimes(0);
    const logout = wrapper.find('button').prop('onClick');
    logout();    
    expect( setUser ).toHaveBeenCalledTimes(1); 
    //expect(JSON.parse(wrapper.find('pre').text().trim())).toEqual({});  
  });

});