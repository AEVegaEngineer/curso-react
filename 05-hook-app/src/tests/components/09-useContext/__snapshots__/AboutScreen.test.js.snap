// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AboutScreen/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    AboutScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Andres",
   "email": "aevega1991@gmail.com"
}
  </pre>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
